src/sagemaker/processing.py

```diff
@@ -1265,9 +1266,8 @@ class FeatureStoreOutput(ApiObject):
 class FrameworkProcessor(ScriptProcessor):
     """Handles Amazon SageMaker processing tasks for jobs using a machine learning framework."""
 
-    framework_entrypoint_command = ["/bin/bash"]
+    framework_entrypoint_command = ["/bin/sh"]
```

Comment:
This change will somewhat restrict the available syntax for the framework script, but that's private logic (`_generate_framework_script`). It won't change anything about the execution environment which is still `python` per line 1346 (should this actually be changed to `python3` maybe?). The main effect it will have on users is compatibility with different container images by standardising on POSIX-compliant `sh` rather than the more advanced `bash` shell which some OSs don't offer out-of-the-box (e.g. [Alpine Linux](https://wiki.alpinelinux.org/wiki/How_to_get_regular_stuff_working) or others mentioned [here](https://stackoverflow.com/a/5725402)). While I'm aware of multiple distros that offer some `sh` implementation but not `bash` in particular, I've not found any that go the other way around.

From my investigations, Ubuntu (16+) is used as a base by all of the [DLCs](https://github.com/aws/deep-learning-containers), the [Scikit-Learn framework container](https://github.com/aws/sagemaker-scikit-learn-container), the [XGBoost framework container](https://github.com/aws/sagemaker-xgboost-container), and even the [Chainer container](https://github.com/aws/sagemaker-chainer-container). As per [here](https://wiki.ubuntu.com/DashAsBinSh), Ubuntu has supported both `/bin/sh` and `/bin/bash` since at least v6.

The [Spark container](https://github.com/aws/sagemaker-spark-container) (which isn't actually FrameworkProcessor-based) seems to use Amazon Linux 2, which also offers `/bin/sh`.

Overall from what I can tell this is not breaking and from a compatibility standpoint should be a step forward rather than back?

Reply:
Thanks, this makes sense